### DISCUSSION TITLE
Datatable: Option to disable column headers when emptyMessage is shown

### DISCUSSION THREAD

**Discussion by gayathrirajendran** [primefaces/community#1333] [PrimeReact]
Hello,

Can you please consider adding the option to disable all the column headers (so that the sort and filters cannot happen) exactly at the time emptyMessage is shown. An overlay is fine too just like loading today.

Thanks!

**Comment by melloware** [comment]
you can do it right now: https://stackblitz.com/edit/4d1epv?file=src%2FApp.jsx

Click the "Clear Table" button

**Comment by gayathrirajendran** [comment]
@melloware The requirement is not to hide the headers totally. It is to just disable sorting filtering etc that happens over network calls, a lazy table.

**Reply by melloware** [reply]
Hmm i am not sure why it matters why is your user trying to sort or filter an empty table?

**Reply by gayathrirajendran** [reply]
Hmm, we never know what users end up trying to do right. Looking to plug all situations which would lead to wasteful interactions and api calls. So a disableOnEmptyOption would help.
